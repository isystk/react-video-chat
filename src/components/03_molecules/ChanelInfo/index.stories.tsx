import { Meta, Story } from '@storybook/react'
import React from 'react'
import ChanelInfo from './index'
import MainService from '@/services/main'

export default {
  title: '03_molecules/ChanelInfo',
  component: ChanelInfo,
} as Meta

const main = {
  chanels: {
    all: {
      id: 'all',
      name: 'すべて',
      type: 'all',
      photo: 'images/friends/Alpha_Team.png',
      detail: 'ルーム内のすべてのメンバー',
      chat: {
        chanelId: 'test',
        messages: [],
      },
    },
  },
  selectChanelId: 'all',
} as MainService

const Template: Story = () => <ChanelInfo main={main} />

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
