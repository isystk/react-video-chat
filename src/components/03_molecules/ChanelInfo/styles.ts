import { css, keyframes } from '@emotion/css'

export const meBlock = css`
  padding: 20px 10px;
  height: 100%;
  & .head-photo {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    margin: 0 auto;
    overflow: hidden;
  }
  & .head-photo img {
    width: 100%;
  }
  & .name {
    width: 100%;
    font-size: 24px;
    color: #222222;
    text-align: center;
    margin-top: 15px;
  }
  & .title {
    width: 100%;
    font-size: 16px;
    color: #777777;
    text-align: center;
    margin-top: 5px;
  }
  & .video {
    width: 100%;
    text-align: center;
    margin-top: 5px;
  }
  & .video a {
    color: #777777;
  }
  & .video svg {
    font-size: 2.5em;
  }
`
