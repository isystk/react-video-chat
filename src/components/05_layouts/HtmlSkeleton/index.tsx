import React, { Children, FC, useState } from 'react'
import { connect } from '@/components/hoc'
import { ContainerProps, WithChildren } from 'types'
import useAppRoot from '@/stores/useAppRoot'
import Header from '@/components/04_organisms/Header'
import SideMenu from '@/components/04_organisms/SideMenu'
import Title, { TitleProps } from './Title'
import Head from 'next/head'
import { isReactElement } from '@/utils/general/object'
import { CssBaseline } from '@material-ui/core'

/** HtmlSkeleton Props */
export type HtmlSkeletonProps = WithChildren
/** Presenter Props */
export type PresenterProps = HtmlSkeletonProps & {
  title?: TitleProps['children']
  description: string
}

/** Presenter Component */
const HtmlSkeletonPresenter: FC<PresenterProps> = ({
  main,
  title,
  description,
  isMenuOpen,
  setMenuOpen,
  children,
  ...props
}) => (
  <>
    <Head>
      {/* タイトル */}
      <title>{title}</title>
      {/* favicon */}
      <link rel="icon" href="/favicon.ico" />
      {/* ブラウザテーマカラー */}
      <meta name="theme-color" content="#ffffff" />
      {/* サイト概要 */}
      <meta name="description" content={description} />
      {/* OGP 画像URL */}
      <meta property="og:image" content={'/ogp-image.png'} />
      {/* OGP タイトル */}
      <meta name="og:title" content={title} />
      {/* OGP サイト概要 */}
      <meta name="og:description" content={description} />
      {/* OGP Twitterカード */}
      <meta name="twitter:card" content="summary" />
      {/* apple ポータブル端末 アイコン */}
      <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
      {/* manifest.json */}
      <link rel="manifest" href="/manifest.json" />
    </Head>
    <Context.Provider value={main}>
      <CssBaseline />
      <Header isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
      <div>{children}</div>
      <SideMenu isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
    </Context.Provider>
  </>
)

/** Container Component */
const HtmlSkeletonContainer: React.FC<
  ContainerProps<HtmlSkeletonProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  let title: TitleProps['children'] | undefined = undefined
  const main = useAppRoot()
  const [isMenuOpen, setMenuOpen] = useState(false)

  if (!main) return <></>

  const APP_NAME = process.env.APP_NAME

  children = Children.map(children, (child) =>
    isReactElement(child) && child.type === Title
      ? (title = `${child.props.children} | ${APP_NAME}`) && undefined
      : React.cloneElement(child)
  )
  if (!title) title = process.env.APP_NAME

  const description = `${APP_NAME} is an application that allows you to casually chat with your friends!`
  return presenter({
    main,
    title,
    description,
    isMenuOpen,
    setMenuOpen,
    children,
    ...props,
  })
}

export const Context = React.createContext(null)

export default connect<HtmlSkeletonProps, PresenterProps>(
  'HtmlSkeleton',
  HtmlSkeletonPresenter,
  HtmlSkeletonContainer
)

// Sub Component
export type { TitleProps }
export { Title }
