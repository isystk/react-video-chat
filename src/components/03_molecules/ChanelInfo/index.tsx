import * as React from 'react'
import VideocamIcon from '@material-ui/icons/Videocam'
import VideocamOffIcon from '@material-ui/icons/VideocamOff'
import * as _ from 'lodash'
import { ContainerProps, WithChildren } from 'types'
import { useStyles } from './styles'
import { FC, useContext } from 'react'
import { connect } from '@/components/hoc'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'

/** ChanelInfoProps Props */
export type ChanelInfoProps = WithChildren
/** Presenter Props */
export type PresenterProps = ChanelInfoProps & {
  sendRequestCall
  classes
  id
  name
  type
  photo
  detail
}

/** Presenter Component */
const ChanelInfoPresenter: FC<PresenterProps> = ({
  main,
  classes,
  id,
  name,
  type,
  photo,
  detail,
  ...props
}) => (
  <>
    <div className="me_block">
      <div className="myHeadPhoto">
        <img src={photo} alt="" />
      </div>
      <div className="myName">{name}</div>
      <div className="myTitle">{detail}</div>
      <div className="myVideo">
        {'other' === type ? (
          <a href="#" onClick={() => main.video.sendRequestCall(id)}>
            <VideocamIcon />
          </a>
        ) : (
          <VideocamOffIcon />
        )}
      </div>
    </div>
  </>
)

/** Container Component */
const ChanelInfoContainer: React.FC<
  ContainerProps<ChanelInfoProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const main = useContext<MainService | null>(Context)
  if (!main || _.size(main.chanels) === 0) return <></>
  const classes = useStyles()

  const { id, name, type, photo, detail } = main.chanels[main.selectChanelId]
  return presenter({
    children,
    main,
    classes,
    id,
    name,
    type,
    photo,
    detail,
    ...props,
  })
}

export default connect<ChanelInfoProps, PresenterProps>(
  'ChanelInfo',
  ChanelInfoPresenter,
  ChanelInfoContainer
)
