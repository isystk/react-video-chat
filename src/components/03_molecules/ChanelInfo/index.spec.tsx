import React from 'react'
import renderer from 'react-test-renderer'
import ChanelInfo from './index'
import { Context } from '@/components/05_layouts/HtmlSkeleton'
import '@testing-library/jest-dom/extend-expect'
import MainService from '@/services/main'
import ChanelService from '@/services/chanel'

describe('ChanelInfo', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    main.setName('isystk')
    main.setRoomId('test')
    main.addChanel(
      new ChanelService(
        main,
        'all',
        'すべて',
        'all',
        'images/friends/Alpha_Team.png',
        'ルーム内のすべてのメンバー'
      )
    )
    main.setChanelId('all')

    const component = renderer.create(
      <Context.Provider value={main}>
        <ChanelInfo />
      </Context.Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
