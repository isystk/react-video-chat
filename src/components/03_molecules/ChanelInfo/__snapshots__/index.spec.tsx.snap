// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChanelInfo Match Snapshot 1`] = `
<div
  className="me_block"
>
  <div
    className="myHeadPhoto"
  >
    <img
      alt=""
      src="images/friends/Alpha_Team.png"
    />
  </div>
  <div
    className="myName"
  >
    すべて
  </div>
  <div
    className="myTitle"
  >
    ルーム内のすべてのメンバー
  </div>
  <div
    className="myVideo"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M21 6.5l-4 4V7c0-.55-.45-1-1-1H9.82L21 17.18V6.5zM3.27 2L2 3.27 4.73 6H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.21 0 .39-.08.54-.18L19.73 21 21 19.73 3.27 2z"
      />
    </svg>
  </div>
</div>
`;
